numba/core/types/functions.py

```diff
@@ -1,58 +1,202 @@
-        msgbuf = [headtmp.format(self._function_type, ', '.join(args))]
-        explain = self._context.explain_function_type(self._function_type)
-        msgbuf.append(explain)
-        for i, (temp, error) in enumerate(self._failures):
-            msgbuf.append(_termcolor.errmsg("In definition {}:".format(i)))
-            msgbuf.append(_termcolor.highlight('{}{}'.format(
-                indent, self.format_error(error))))
-            loc = self.get_loc(temp, error)
+        indent = ' ' * self._scale
+        argstr = argsnkwargs_to_str(self._args, self._kwargs)
+        ncandidates = sum([len(x) for x in self._failures.values()])
+
+        # sort out a display name for the function
+        tykey = self._function_type.typing_key
+        # most things have __name__
+        fname = getattr(tykey, '__name__', None)
+        is_external_fn_ptr = isinstance(self._function_type,
+                                        ExternalFunctionPointer)
+        if fname is None:
+            if is_external_fn_ptr:
+                fname = "ExternalFunctionPointer"
+            else:
+                fname = "<unknown function>"
+
+        msgbuf = [_header_template.format(the_function=self._function_type,
+                                          fname=fname,
+                                          signature=argstr,
+                                          ncandidates=ncandidates)]
+        nolitargs = tuple([unliteral(a) for a in self._args])
+        nolitkwargs = {k: unliteral(v) for k, v in self._kwargs.items()}
+        nolitargstr = argsnkwargs_to_str(nolitargs, nolitkwargs)
+
+        key = self._function_type.key[0]
+        fn_name = getattr(key, '__name__', str(key))
+
+        # depth could potentially get massive, so limit it.
+        ldepth = min(max(self._depth, 0), self._max_depth)
+
+        for i, (k, err_list) in enumerate(self._failures.items()):
+            err = err_list[0]
+            nduplicates = len(err_list)
+            template, error = err.template, err.error
+            source_fn = template.key
+            if isinstance(source_fn, numba.core.extending._Intrinsic):
+                source_fn = template.key._defn
+            elif (is_external_fn_ptr and
+                  isinstance(source_fn, numba.core.typing.templates.Signature)):
+                source_fn = template.__class__
+            source_file, source_line = self._get_source_info(source_fn)
+            largstr = argstr if err.literal else nolitargstr
+            msgbuf.append(_termcolor.errmsg(
+                _wrapper(_overload_template.format(nduplicates=nduplicates,
+                                                  function=source_fn.__name__,
+                                                  file=source_file,
+                                                  line=source_line,
+                                                  args=largstr),
+                ldepth + 1)))
+            if error is None:
```

Comment:
i guess this whole branch can be removed in this PR.

Reply:
this is the same as https://github.com/numba/numba/pull/5722#discussion_r430755865

Reply:
done